doc/lua_api.txt

```diff
@@ -1498,7 +1498,26 @@ Exact pointing location (currently only `Raycast` supports these fields):
   Is a null vector `{x = 0, y = 0, z = 0}` when the pointer is inside the
   selection box.
 
+`move_result`
+---------------
 
+* `{collides_xz, collides_y, standing, collisions}`
```

Comment:
Way too many utils, information duplication.

Reply:
> Way too many utils, information duplication.

This is not "information duplication", rather it is a summary report of collision processing. Having to manually loop over the collisions table to determine whether an object is standing on the ground or colliding on the XZ or Y axes is not only excessive, but it introduces greater likelihood of errors and oversights. 

There is absolutely nothing wrong with reporting summary information that can be otherwise obtained algorithmically. Proof of this is that the official API itself often "duplicates" information that can be ascertained through multiple different means.

Here are plenty of examples that defy both of your arguments:

**Information Duplication:**
* We have a "name" field exposed to LuaEntities, even though there's already a `get_entity_name()` method providing the same information. Yet ironically, `get_entity_name()` was officially deprecated, thus forcing the use of a highly volatile field that can be manipulated by any negligent mod, making it altogether unreliable anyway.

* We have a `minetest.player_exists()` function, which is equivalent to calling the `get_auth()` method of the registered authentication handler and checking if the return value is a table, in which case the player exists.

**Way Too Many Utils:**
* We have a `minetest.get_connected_players()` function in CPP, even though it's very easy to track which players are online using the on_joinplayer and on_leaveplayer callbacks. In fact, this is the methodology in my Player Registry mod. So not only is this API function redundant, since it can be easily implemented as a mod, but it also poorly named since it doesn't check if the player is "connected", but instead whether the player is in the environment. 

* We have a `minetest.get_uptime()` function in CPP, even though mods can already compute the uptime interval of the server by calling `os.time()` at startup, and then again at any future moment to determine the difference. So there is no need for this utility function to be included in the engine since it can be implemented as a mod.

* We have a `minetest.get_player_by_name()` function, even though mods can easily maintain their own cross-reference of player names to ObjectRefs. In fact, I provide such a public lookup table in my Player Registry mod. So there is no need for this utility function to be included in the engine since it can be implemented as a mod.

* We have an `is_player()` method of ObjectRefs, even though it's easy to ascertain whether objects in the environment are players by maintaining a hash table in conjunction with the on_joinplayer and on_leaveplayer callbacks. So there is no need for this utility function to be included in the engine since it can be implemented as a mod.

* We have a `minetest.debug()` function that does absolutely nothing more than concatenate a table to pass directly `minetest.log()`. But I guess it's perfectly fine when the official API has these types of shorthand utility functions. But heaven forbid I provide any of my own, then it becomes "too many utils" for people to handle.

* We have `table.shuffle()` function, even though I have yet too see any widespread application for such a feature, other than sheer novelty. The author claims it can randomize a chest inventory. Yet, most players would prefer their chests be organized. If "too many utils" is objectionable, then, it's strange why this was ever approved.

* We have a `minetest.setting_get_pos()` function that defies the convention of obtaining settings via the Settings object. Wrapping such a function into Settings where it belongs would make for a logically consistent API.  And that's not what we want in Minetest, unless it comes to this PR. Then suddenly perfectionism is demanded.

* We have a `vector.new()` function, which does nothing more than return a table in the format {x,y,z}. That takes more keystrokes and is just an extra function call for no substantial gain in either readability or efficiency. So there is no need for this utility function to be included in the engine since it can be implemented within a mod.

* We have entire VoxelArea class, which is nothing but one gigantic utility for simplifying voxel area calculations. It's only purpose is to make it easier for novice coders that cannot grasp how to translate 3D node coordinates to a linear array index. This utility class does not belong in the engine, since it can be implemented as a mod.

* Of course the prime example of a poorly thought out API is the convoluted falling nodes "feature" that is baked into the engine. It is a highly specific game mechanic consisting of 600 lines of code for a specific use-case. For years I've been calling for falling nodes to be removed from the core and implemented as a standalone mod.

* It's not just falling nodes either. Item entities also have no place in the engine. They increase code complexity, yet are not generic enough to be useful outside of a single type of game. There's almost no allowance for customization, since everything is hard-coded. This encourages undocumented hacks to override builtin functionality.

Rest assured, I could have a field day going down my long list of how badly designed the Minetest API is. So many of these shortcomings are glaringly obvious too. But instead of addressing the existing API that modders are forced to work with, you object to this PR with little to no evidence, only incoherent remarks like "poor code quality".

So many of your arguments aren't even grounded in reason, like the fallacious claim that this PR causes overhead for non-mob entities. I'm willing to bet you didn't run a single test to determine whether that was the case. Then of course, there is your assertion that new_vel and old_vel are provided by the current move_result, which I've already debunked.

In fact, not ONE criticism of this PR have you provided useful advice or direction. It's only been outright complaints. Is that how you make worthwhile contributions to an open-source project by blasting proposals that you don't favour (for obviously personal reasons) with no constructive input whatsoever? 

It's abundantly clear that your overall rebuke of this PR is not in good faith. Rather you see it as a competition with your "Moblib" mod. And that is neither worthwhile nor helpful to anybody. Modders should not have to depend on your library just to perform basic entity manipulations and calculations that can be handled effectively by the engine. That is further evidence that the Minetest API is woefully inadequate.